### ISSUE TITLE
Pagination Limit does not work

### ISSUE THREAD

**Issue by vexersa** [mautic/mautic#13553]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1

### What browsers are you seeing the problem on?

Firefox, Chrome

### What happened?

Selecting pagination limit on any page has no effect and resets value to default.

### How can we reproduce this issue?

1. Open page with multiple items, such as Custom Fields 
2. Select another option for pagination limit
3. Nothing happens and the selection resets to 30 (default)

### Relevant log output

```shell
Nothing is produced in the logs.
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by dsp76** [comment]
Confirmed for the custom fields settings in 5.0.4 - it was working at least in 4.4.12. 
@vexersa  - does it work in contact, company lists? Which other lists having the same problem?

**Comment by vexersa** [comment]
@dsp76 

I've just tested and no fields sort when clicked, custom fields or otherwise.

**Comment by dsp76** [comment]
I mean - its a confirmed bug. Did you also test lists in other parts of Mautic 5 in regards to pagination? 

(The Bug is about pagination, not about sorting.  Sorting was not implemented for custom fields anyway. Sorting is working in other lists within Mautic 5, like contact, companies, assets etc.)

**Comment by vexersa** [comment]
Apologies, I mixed up the bug with sorting and this issue. [This one.](https://github.com/mautic/mautic/issues/13313)

This pagination issue only seems to happen on the Contacts page.

**Comment by PauperAnteFestum** [comment]
Unfortunately, they have released version 5.04 and have not addressed this issue. Sorting the contact list is extremely important; how can it not be resolved?

**Comment by RCheesley** [comment]
Hi folks, I've just been reviewing this in the [5.x branch](https://gitpod.io/#https://github.com/mautic/mautic) with sample data on the Contacts screen, and I am not able to reproduce the issue.

Pagination works fine, and I'm able to filter to 5, 15 etc and both pagination and sorting are working fine.

https://app.screencastify.com/v3/watch/4F5kh9WO3StzLgKOI6fC

If this isn't the same issue you're reporting please let me know and we can re-open the issue.

I am going to close this issue, if you are still having a problem I suggest:

1. Clearing the cache
2. Regenerating your CSS/JS assets using `bin/console m:a:g`

**Comment by pjrobertson** [comment]
I can confirm this is fixed. In the future, it would be really helpful for other users to list the commit that fixes a bug. That way, users can patch/pull that single commit to fix the issue without having to wait for a new release!

For those coming across this issue, the commit that fixed the problem is cc0027778a911ca22837d52944571ae571cbaadc

If you want to fix this now without waiting for 5.x, then you can run the following:

```bash
curl https://github.com/mautic/mautic/commit/cc0027778a911ca22837d52944571ae571cbaadc.patch | sudo patch -p1
bin/console mautic:a:g
```
